services-api/src/main/java/io/scalecube/services/ServiceProvider.java

```diff
@@ -2,7 +2,13 @@
 
 import java.util.Collection;
 
+/**
+ * Provide service instances.
+ *
+ * @deprecated use {@link ServicesProvider}
+ */
 @FunctionalInterface
+@Deprecated
```

Comment:
All is cool with this interface. It shouldn't be deprecated. I still want to use good-old approach with `.services(Object... ) / .services(ServiceInfo... )` or `services(ServiceProvider)`. Keep `deprecated` mentioned top level functions on  Microservices (or how u call it here _Scalecube_), but decalre them on the `ScaleCubeServicesProvider` (with the same fluent API style).

Reply:
Sometimes you have to throw away the old one and move on :)  And seriously, this API is deprecated for the following reason: we delegate tasks to store and manage service life cycle to ServiceFactory (ex ServicesProvider) implementations. Scalecube (Microservices) should not perform such tasks, imho. Especially if we allocate a separate component that solves these tasks, I see no sense in leaving the old API, it's better to transfer these methods to the new one. As an option, you can add `registerDefinition(ServiceInfo serviceInfo)` and `<T> registerDefinition(Supplier<T> instanceSupplier, Class<T> type, Tag... tags)` methods to ServiceFactory (ex ServiceProducer). 

In the end.
- I find it wrong to leave the API `.services(ServiceInfo... services)` in Scalecube (Microservices);
- Take this API into the interface and implement it.

P.S. During development, I added the registration of additional services through `register(ServiceInfo service)` but decided to remove it because it violated the immunity of the ServiceFactory (ServicesProvider).

Reply:
its ok to deprecate  .services(ServiceInfo... services) but we cant just do that without discussing the public api.
services is used in all the examples and documentations and we need to direct the users as part of this pull request to a proper use of the api with examples / readme / website and basically everywhere.

plust we are changing this method we also need to discuss the public api from end user point of view. we need to review it to make sure its clean and easy to use.

Reply:
@ronenhamias So api is not removed, but @Deprecated :)

Reply:
> we delegate tasks to store and manage service life cycle to ServiceFactory (ex ServicesProvider) implementations

Correct. We don't want `.services(Object... ) / .services(ServiceInfo... ) or services(ServiceProvider)` on `Microservices` anymore.  That's why:
>  Keep `@deprecated`  functions `.services(Object... ) / .services(ServiceInfo... ) or services(ServiceProvider)` on 

Next.
> As an option, you can add registerDefinition(ServiceIn ...

We don't need options. We need clear and well understood approach. Shall we discuss it here? No. Not in the scope of this PR. That's why: 
> Declare them on the `ScaleCubeServicesProvider` (with the same fluent API style).


Upon this one.
> I see no sense in leaving the old API, it's better to transfer these methods to the new one. 

Good. But not in this PR. In this PR we will keep them but just move to another place.